readme/docker-based-deployment.md

```diff
@@ -0,0 +1,37 @@
+1. Install docker and docker-compose
+2. Build the images 
+   ```sh
+    ./docker/build.sh
+    ```
+3. There are four ways of setting/deploying the resource server using docker:
+   1. Non-Clustered setup with all verticles running in a single container: 
+   - This needs no hazelcast, zookeeper, the deployment can be done on non-swarm too and suitable for development environment.
+   - This makes use of iudx/rs-dev:latest image and config-dev.json present at `secrets/credentials/config-dev.json`
+   ```sh 
+   # Command to bring up the non-clustered resource-server container
+   docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d
```

Comment:
Rather than this, I would prefer shell scripts that do whatever you want inside it.

Reply:
> Rather than this, I would prefer shell scripts that do whatever you want inside it.

@rraks 
Its the docker-compose way of deploying things in different environments using [multiple-compose files](https://docs.docker.com/compose/extends/#multiple-compose-files). 
Moreover scripting is limited, cannot differentiate , for example between attaching docker swarm  overlay-network and local bridge network needed for prod/test and dev environment respectively.
I feel its looks neater, u have base compose and overriding compose file which overrides based on environment.

Reply:
No you can do docker compose only. 
Putting that command in a shell script is better than mentioning it somewhere.
That way you can abstract out functionality from syntax.

Reply:
> No you can do docker compose only.
> Putting that command in a shell script is better than mentioning it somewhere.
> That way you can abstract out functionality from syntax.

The command is same for all : `bash -c "exec java $$RS_JAVA_OPTS  -jar ./fatjar.jar  --host $$(hostname) -c secrets/credentials/all-verticles-configs/config.json`
 except I am acknowledging the two type of configs:  one for clustered (config-dev.json) and another for non-clustered setup (config-depl.json) but both are mounted on same mountpoint inside the docker.

Reply:
> Putting that command in a shell script is better than mentioning it somewhere.
> That way you can abstract out functionality from syntax.

Yes I can put the command into a script, and have to either mount the script or have to burn into image. 
 both has their own demerits. 

But will it not be simple just to mention the one line command in the compose file itself.